build.gradle.kts

```diff
@@ -300,4 +319,47 @@ fun PublicationContainer.setDefaultPomFields() {
       }
     }
   }
+}
+
+tasks.register("publishIfNeeded") {
+  val eventName = System.getenv("GITHUB_EVENT_NAME")
+  val ref = System.getenv("GITHUB_REF")
+  println("publishIfNeeded eventName=$eventName ref=$ref")
+
+  if (eventName == "push" && ref == "ref/heads/master") {
+    println("Deploying snapshot to OSS...")
```

Comment:
nit: I thought Gradle project provides logger, should we use proper `project.logger`?

Reply:
Yep, that'll allow users to shut down the logs if they really want to. I should certainly do this inside the Gradle plugin as well.

Reply:
For the record, I just checked the apollo-gradle-plugin module and it doesn't print to stdout at all so we're all good 🎉